core/sql/optimizer/RelExpr.cpp

```diff
@@ -6630,6 +6630,23 @@ const NAString Intersect::getText() const
 }
 
 // -----------------------------------------------------------------------
+// // member functions for class Except
+// // -----------------------------------------------------------------------
+Except::Except(RelExpr *leftChild,
+             RelExpr *rightChild)
+: RelExpr(REL_EXCEPT, leftChild, rightChild)
+{ setNonCacheable(); }
```

Comment:
Why is this noncacheable?
Also please check if we can do showshape on a statement with EXCEPT